### DISCUSSION TITLE
Mac Performa 460 (LCIIIish) - image created with truncate not recognized

### DISCUSSION THREAD

**Discussion by cmigliorini** [ZuluSCSI/ZuluSCSI-firmware#430] [Q&A] [answered]
I finally succeeded to create an image from my Ubuntu 22 box to be accepted by my Mac 460 and ZuluSCSI 2040, but neither of the following methods worked=
- the `truncate` method indicated in the documentation
- `truncate` + `mkfs.hfs` 
- `truncate` + `fdisk` + `mkfs.hfs` 

The only method I found working was to use [Disk Jockey Junior](https://diskjockey.onegeekarmy.eu/djjr/). Is it just me?

I'm willing to propose a PR to improve documentation.

Note: ZuluSCSI log indicated "WARNING: This image does not appear to be a valid Macintosh disk image."

**Comment by aperezbios** [comment, marked as answer]
@cmigliorini thanks for the report. Real 68k/PPC Macintosh computers can not boot from any SCSI drive that does not contain a valid Apple Partition Map. The boot ROMs in Classic Macs have no awareness of DOS/MBR-style partitioning, so you would have needed to create a partition map with hfdisk, not fdisk. Hence the warning that "This image does not appear to be a valid Macintosh disk image"

I say "any SCSI drive" because Classic Macs booted from the **floppy** interface _can_ boot from a straight HFS disk/disk image, but not the other way around. When you use Disk Jockey/DJJR, it's creating the missing APM for you, which is why it works.

Does this make sense? I'm open to any suggestions for improvement of the documentation.

**Comment by cmigliorini** [comment]
@aperezbios thanks so much for the detailed response. What I did exactly is
- I not only created the hard disk image: I also installed a CD image (System 7.6), and the system properly booted from the CD, which was what I wanted to achieve
- but the installation program did not recognize any hard drive

So you're telling me it would have worked if I had used hfdisk (I should have thought of that, still) to partition my `truncate`d file? I'll give it a try.

At any rate, if this is the solution it'd be of great help that the documentation reflects this. Let me know if I can help.

**Comment by aperezbios** [comment]
@cmigliorini There are two separate issues/things to know here. One is that an image must contain a valid Apple Partition Map, to be able to be recognized by the ROM a classic Macintosh. 

The other is that, by default, all of the original Apple-provided disk formatting utilities will ignore any non-Apple-firmware SCSI hard drive, by design. If you set System=Mac in the zuluscsi.ini file, the ZuluSCSI firmware will report back with device/vendor strings that Apple's un-patched utility (either HD SC Setup, or Drive Setup) should be happy with, at which point it will let you initialize the drive as if it were an OEM Apple HDD. 

Alternatively, you can patch, or use a pre-patched version, of HD SC Setup or Drive Setup utilities, which removes this artificial limitation from the utility, allowing them to initialize any third-party HDD. There's a link to some more info about this at https://en.wikipedia.org/wiki/Apple_HD_SC_Setup

**Reply by cmigliorini** [reply]
Indeed, I understand, so here is what I did
- created an image with `truncate --size 2047M` (exactly the size I used with Disk Jockey Junior)
- using [hfdisk](https://github.com/rabbitholecomputing/hfdisk) (a7659ed4) which I compiled with gcc 12 on my Jammy, I initialized a partition map, which gives
```
/tmp/applemap
             #                    type name                length   base    ( size )  system
/tmp/applemap1     Apple_partition_map Apple                   63 @ 1       ( 31.5k)  Partition map
/tmp/applemap2              Apple_Free Extra              4192192 @ 64      (  2.0G)  Free space
```
- renamed to `HD00.img`, copied to a FAT32 partition on my SD, alongside with a `CD10.iso` containing System 7.6 (FR, because I wanted it to speak french)
- added a `zuluscsi.ini` containing:
```
[SCSI]
# Settings that apply to all SCSI IDs

# Select a system preset to apply default settings
# Known systems: "Mac", "MacPlus", "MPC3000", "MegaSTE", "X68000"
System="Mac"
```
- booted the Macintosh, which rebooted once (?) and properly booted on the CD ROM.. `zululog.txt` yields
```
[12ms] Platform: ZuluSCSI RP2040
[12ms] FW Version: 24.05.17-release May 17 2024 16:24:20
[13ms] DIP switch settings: debug log 0, termination 1
[13ms] SCSI termination is enabled
[14ms] Flash chip size: 4096 kB
[14ms] SCSI target/disk mode selected by DIP switch, acting as a SCSI disk
[38ms] SD card detected, FAT32 volume size: 7564 MB
[38ms] SD MID: 0x03, OID: 0x53 0x44
[39ms] SD Name: SU08G
[39ms] SD Date: 8/2013
[39ms] SD Serial: 0x400A6E9D
[63ms] Reading configuration from zuluscsi.ini
[86ms] Active configuration (using system preset "Mac"):
[86ms] -- SelectionDelay = 255
[87ms] -- EnableUnitAttention = No
[87ms] -- EnableSCSI2 = Yes
[87ms] -- EnableSelLatch = No
[88ms] -- MapLunsToIDs = No
[88ms] -- EnableParity = Yes
[253ms] Finding images in directory /:
[271ms] -- Opening /HD00.img for id:0 lun:0
[4007ms] ---- Configuring as disk drive drive
[4008ms] ---- Read prefetch enabled: 8192 bytes
[4026ms] -- Opening /CD60-MacOS76_FR.iso for id:6 lun:0
[4308ms] ---- Configuring as CD-ROM drive
[4308ms] ---- Read prefetch enabled: 8192 bytes
[4316ms] -- Platform supports ROM drive up to 3740 kB
[4316ms] ---- ROM drive image not detected
[4316ms] SCSI ID: 0, BlockSize: 512, Type: 0, Quirks: 1, Size: 2096128kB
[4317ms] SCSI ID: 6, BlockSize: 2048, Type: 2, Quirks: 1, Size: 159024kB
[6492ms] Initialization complete!
[23228ms] Toolbox enabled = 0
```
- but then the "install Mac OS" software does not find any hard drive to "upgrade" or install to
